lib/zipEntry.js

```diff
@@ -203,13 +215,21 @@ ZipEntry.prototype = {
         while (reader.index < end) {
             extraFieldId = reader.readInt(2);
             extraFieldLength = reader.readInt(2);
-            extraFieldValue = reader.readData(extraFieldLength);
 
-            this.extraFields[extraFieldId] = {
-                id: extraFieldId,
-                length: extraFieldLength,
-                value: extraFieldValue
-            };
+            // AES encrypt extra field header id
+            if(extraFieldId === 0x9901){
+                this.options.aes.version = reader.readInt(2);
+                reader.skip(2);
+                this.options.aes.strength = reader.readInt(1);
+                this.compressionMethod = this.options.aes.compressionMethod = reader.readString(2);
+            }else{
```

Comment:
```suggestion
            } else {
```

Please also change the other places where this does not follow the existing convention.